### PR TITLE
Add done not done

### LINKED ISSUE #62: Implement Done/Not Done feature

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
Fixes #62

### CODE REVIEW

File: src/test/data/JsonModuleBookStorageTest/invalidAndValidTaskModuleBook.json [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
+    "doneStatus": "false"
   }, {
     "name": "Task With Invalid Deadline Field",
     "deadline": "948asdf2424",
     "module": "hans@example.com",
-    "description": "4th street"
+    "description": "4th street",
+    "doneStatus": "false"
   } ]
 }
```
Comment (QY-H00) [inline src/test/data/JsonModuleBookStorageTest/invalidAndValidTaskModuleBook.json:15]: Here the value of property isn't change to correct format of MB3.5.

File: src/test/data/JsonSerializableModuleBookTest/invalidTaskModuleBook.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "name": "Hans Muster",
     "deadline": "9482424",
     "module": "invalid@module!3e",
-    "description": "4th street"
+    "description": "4th street",
+    "doneStatus": "false"
   } ]
 }
```
Comment (QY-H00) [inline src/test/data/JsonSerializableModuleBookTest/invalidTaskModuleBook.json:9]: Here the value of property isn't change to correct format of MB3.5.

File: src/test/java/seedu/module/storage/JsonAdaptedTaskTest.java [hunk 3]
```diff
@@ -14,20 +14,23 @@
...
     private static final String INVALID_TAG = "#friend";
 
     private static final String VALID_NAME = MIDTERM.getName().toString();
     private static final String VALID_DEADLINE = MIDTERM.getDeadline().toString();
     private static final String VALID_MODULE = MIDTERM.getModule().toString();
     private static final String VALID_ADDRESS = MIDTERM.getDescription().toString();
+    private static final String VALID_DONE_STATUS = MIDTERM.getDoneStatus().toString();
     private static final List<JsonAdaptedTag> VALID_TAGS = MIDTERM.getTags().stream()
             .map(JsonAdaptedTag::new)
             .collect(Collectors.toList());
```
Comment (QY-H00) [inline src/test/java/seedu/module/storage/JsonAdaptedTaskTest.java:36]: Here we also need to fix the property to MB3.5 to make it a meaningful test although it's invalid test.

**Review by QY-H00 — COMMENTED** [review]
LGTM. Just some value of json file need to match MB3.5.

File: src/test/data/JsonModuleBookStorageTest/invalidAndValidTaskModuleBook.json (hunk 1 above)
Comment (yhtMinceraft1010X) [reply src/test/data/JsonModuleBookStorageTest/invalidAndValidTaskModuleBook.json:15]: To clarify, the module field should be changed to a proper module code?

File: src/test/data/JsonSerializableModuleBookTest/invalidTaskModuleBook.json (hunk 2 above)
Comment (yhtMinceraft1010X) [reply src/test/data/JsonSerializableModuleBookTest/invalidTaskModuleBook.json:9]: To clarify, is the deadline here supposed to change?

File: src/test/java/seedu/module/storage/JsonAdaptedTaskTest.java (hunk 3 above)
Comment (yhtMinceraft1010X) [reply src/test/java/seedu/module/storage/JsonAdaptedTaskTest.java:36]: I assume here the ADDRESS should be changed to DESCRIPTION. Is there any other changes to make?

**Review by figo2127 — COMMENTED** [review]
LGTM overall.

**Review by AiwassPrime — COMMENTED** [review]
Do you think a test for ModuleBookParser is needed here?
Else than that looks good. Nice work!

File: src/test/data/JsonModuleBookStorageTest/invalidAndValidTaskModuleBook.json (hunk 1 above)
Comment (umergta) [reply src/test/data/JsonModuleBookStorageTest/invalidAndValidTaskModuleBook.json:15]: Yes I believe so

**Comment by umergta** [comment]
Overall LGTM. Just added some comments pertaining to previous ongoing convos.

**Pull request merged**